docs/README-agent-install.md

```diff
@@ -164,318 +153,303 @@ Virtual service guestshell+ detail
+
+**Example**
+
+~~~
+node default
+  include ciscopuppet::install
+  class {'ciscopuppet::install':
+    repo  => 'http://gemserver.domain.com:8808',
+    proxy => 'http://proxy.domain.com:8080',
+  }
+end
 ~~~
 
-See the following references for more puppet.conf settings:
+##### Gem Persistence
+
+Once installed, the GEM will remain persistent across system reloads within the Guestshell or OAC environments; however, the bash-shell environment does not share this persistent behavior, in which case the `ciscopuppet::install` helper class automatically downloads and re-installs the gem after each system reload.
+
+* The gem can also be manually installed on the agent node
 
-* https://docs.puppetlabs.com/puppet/latest/reference/config_important_settings.html
-* https://docs.puppetlabs.com/puppet/latest/reference/config_about_settings.html
-* https://docs.puppetlabs.com/puppet/latest/reference/config_file_main.html
-* https://docs.puppetlabs.com/references/latest/configuration.html
+~~~
+gem install cisco_node_utils
+~~~
+<br>
 
-#### Run the Puppet Agent
+#### *Step 4. Run Puppet Agent*
 
 ~~~bash
 puppet agent -t
 ~~~
+<br>
 
-## <a name="ha">Guestshell & High Availability (HA) Platforms</a>
+## <a name="ha">Optional Setup: Guestshell on High Availability (HA) Platforms</a>
 
-Optional. This section discusses `guestshell` usage on HA platforms. This section does not apply to the bash-shell environment, open agent container (OAC) environment or to single-sup platforms.
+This section discusses `guestshell` usage on HA platforms. This section does not apply to the bash-shell environment, open agent container (OAC) environment or to single-sup platforms.
 
 The `guestshell` container does not automatically sync filesystem changes from the active processor to the standby processor. This means that Puppet installation files and related file changes performed in the earlier steps will not be present on the standby until they are manually synced with the following NX-OS exec command:
 
 ~~~
 guestshell sync
 ~~~
+<br>
 
-## <a name="persistence">Puppet Agent Persistence</a>
-
-Optional. This section discusses Puppet agent persistence after system restarts.
+## <a name="persistence">Optional Setup: Puppet Agent Persistence</a>
 
-1. [Service Management in bash-shell using init.d](#svc-mgmt-bs)
-2. [Service Management in guestshell using systemd](#svc-mgmt-gs)
-3. Not supported for the open agent container (OAC)
+This section discusses Puppet agent persistence after system restarts.
 
 #### Service Management
 
 It may be desirable to set up automatic restart of the Puppet agent in the event of a system reset. The bash and guestshell environments use different methods to achieve this.
 
-#### <a name="svc-mgmt-bs">Optional: bash-shell / init.d</a>
+* [Service Management in bash-shell using init.d](#svc-mgmt-bs)
+* [Service Management in guestshell using systemd](#svc-mgmt-gs)
+* The open agent container (OAC) does not support agent persistence
```

Comment:
* No systemd in OAC
* init.d is there but it hasn't been tested

I'll reword the statement to reflect no official support for it.